File: Review-06-Python-Shapes/Example-1/tests/shapes/test_shape_factory.py
from hamcrest import *
import unittest

from shapes.Shape import (Shape, FPT_FMT)
from shapes.ShapeFactory import ShapeFactory

import copy


class TestShapeFactory(unittest.TestCase):
    """
    **This is technically an Integration Tests**

    1 - Does this piece of code perform the operations
        it was designed to perform?

    2 - Does this piece of code do something it was not
        designed to perform?

    1 Test per mutator
    """

    def setUp(self):
        self.known_shapes = copy.deepcopy(ShapeFactory._known_shapes)
        self.number_known = len(self.known_shapes)

    def test_create_shape_success(self):
        """
        Create a known valid Shape
        """

        assert_that(ShapeFactory.create("Circle"), is_not(None))

    def test_create_shape_failure(self):
        """
        Try to create a known invalid Shape
        """

        assert_that(ShapeFactory.create("Lol Nope"), is_(None))

    def test_is_known_success(self):
        """
        Create a known valid Shape
        """

        assert_that(ShapeFactory.is_known("Circle"), is_(True))

    def test_is_known_failure(self):
        """
        Try to create a known invalid Shape
        """

        assert_that(ShapeFactory.is_known("Lol Nope"), is_(False))

    def test_list_known(self):
        known_str = ShapeFactory.list_known()

        for pair in self.known_shapes:
            assert_that(known_str, contains_string(pair.name))

    def test_number_known(self):
        assert_that(ShapeFactory.number_known(), equal_to(self.number_known))

File: 06-Java-Shapes/Example-2/src/test/java/shapes/TestShapeFactory.java
package shapes;

import org.junit.jupiter.api.TestMethodOrder;
import org.junit.jupiter.api.MethodOrderer;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.BeforeEach;

import static org.junit.jupiter.api.Assertions.*;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.*;
import org.hamcrest.core.IsNull;
import static org.hamcrest.number.IsCloseTo.closeTo;

import java.util.List;
import java.util.Arrays;

/**
 *   **This is technically a set of Integration Tests**
 *
 * 1 - Does this piece of code perform the operations
 *     it was designed to perform?
 *
 * 2 - Does this piece of code do something it was not
 *     designed to perform?
 *
 * 1 Test per mutator
 */
@TestMethodOrder(MethodOrderer.MethodName.class)
public class TestShapeFactory {

    private List<String> knownNames;

    private int numberKnown;

    @BeforeEach
    public void setUp()
    {
        this.knownNames = Arrays.asList("Circle",
                                        "Square",
                                        "Triangle",
                                        "Equilateral Triangle",
                                        "Right Triangle");

        this.numberKnown = this.knownNames.size();
    }

    /**
     * Create a known valid Shape
     */
    @Test
    public void testCreateShapeSuccess()
    {
        assertThat(ShapeFactory.createShape("Circle"), is(notNullValue()));
    }

    /**
     * Try to create a known invalid Shape
     */
    @Test
    public void testcreateShapeFailure()
    {
        assertThat(ShapeFactory.createShape("Lol Nope"), is(nullValue()));
    }

    /**
     * Create a known valid Shape
     */
    @Test
    public void testIsKnownSuccess()
    {
        assertThat(ShapeFactory.isKnown("Circle"), is(true));
    }

    /**
     * Try to create a known invalid Shape
     */
    @Test
    public void testIsKnownFailure()
    {
        assertThat(ShapeFactory.isKnown("Lol Nope"), is(false));
    }

    @Test
    public void testListKnown()
    {
        String knownStr = ShapeFactory.listKnown();

        for (String name : this.knownNames) {
            assertThat(knownStr, containsString(name));
        }
    }

    @Test
    public void testNumberKnown()
    {
        assertThat(ShapeFactory.numberKnown(), equalTo(this.numberKnown));
    }
}
